src/api/hooks/useApiQuery.ts

```diff
@@ -12,9 +12,10 @@ function useApiQuery<TData>(
   options: UseQueryOptions<TData> = {enabled: true},
 ) {
   const {api} = useApi();
+  const network = api?.runtimeChain != null ? api.runtimeChain.toString() : '';
 
   return useQuery(
-    queryKey,
+    `${queryKey}-${network}`,
```

Comment:
just one thing, the `queryKey` can be a string or array of string and objects. how about doing something like
```
[network, ...(typeof queryKey === 'string' ? [queryKey]: queryKey)]
```